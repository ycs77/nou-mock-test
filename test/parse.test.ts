import fs from 'node:fs'
import path from 'node:path'
import { expect, it } from 'vitest'
import { pdfFiles } from './data/pdf-files'

it.each(pdfFiles)('parse pdf string to exam data - %s', async filename => {
  const content = fs.readFileSync(path.resolve(__dirname, `fixtures/${filename}__2_loaded_from_pdf.txt`), 'utf-8')

  const blocks = parseExam(content)

  // console.log(require('node:util').inspect(blocks, false, null, true))

  expect(blocks).matchSnapshot()
})
