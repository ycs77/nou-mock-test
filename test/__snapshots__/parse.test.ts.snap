// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse pdf string to exam data - 1011_760008_11 1`] = `
[
  {
    "subject": "國立空中大學 101 學年度上學期期中考試題",
    "type": "title",
  },
  {
    "subject": "科目：Linux 作業系統管理",
    "type": "subtitle",
  },
  {
    "children": [
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "P.3",
        "subject": "1. Linux作業系統是屬於開放原始碼且為免費的自由軟體，其軟體無任何授權限制。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "P.9",
        "subject": "2. Linux作業系統具有多人多工的優點。",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "P.13",
        "subject": "3. Ubuntu 作業系統的正體中文官網提供了最新版本作業系統軟體下載，其網址為http://www.ubuntu-tw.com",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "P.20",
        "subject": "4. Linux作業系統的目錄結構為樹狀結構，其根目錄為 /root。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "P.89",
        "subject": "5. Ubuntu中執行apt-get指令時，必須啟動管理者權限。",
        "type": "radio",
      },
    ],
    "scoreOfItem": 4,
    "scoreTotal": 20,
    "subject": "一、是非題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": "B",
        "options": [
          "(A) /dev/hd[a-z]",
          "(B) /dev/sd[a-z]",
          "(C) /dev/fd[a-z]",
          "(D) /dev/ud[a-z]",
        ],
        "reference": "P.18",
        "subject": "1.下列何者為Linux作業系統正確的USB隨身碟的裝置代號[範圍]?",
        "type": "radio",
      },
      {
        "answer": "D",
        "options": [
          "(A) IPod",
          "(B) IPus",
          "(C) IPad",
          "(D) IBus",
        ],
        "reference": "P.56 圖2-11",
        "subject": "2.在Ubuntu中，其輸入法的設定工具名稱為",
        "type": "radio",
      },
      {
        "answer": "D",
        "options": [
          "(A) ~代表的是「家目錄」",
          "(B) HP-600為主機名稱",
          "(C) $為一般使用者提示字元",
          "(D) ubuntu為root帳號",
        ],
        "reference": "P.114",
        "subject": "3.開啟Ubuntu中的終端機時，若出現的前導字為「ubuntu@HP-600:~$」，下列敘述何者為「非」?",
        "type": "radio",
      },
      {
        "answer": "D",
        "options": [
          "(A) cat",
          "(B) more",
          "(C) tac",
          "(D) top",
        ],
        "reference": "P.131",
        "subject": "4.下列何者指令與顯示檔案內容無關?",
        "type": "radio",
      },
      {
        "answer": "C",
        "options": [
          "(A) 10",
          "(B) 15",
          "(C) -15",
          "(D) -25",
        ],
        "reference": "P.167",
        "subject": "5.若某一執行程序之nice值為0，若要將其執行優先順序調高，則應將nice值調為?",
        "type": "radio",
      },
    ],
    "scoreOfItem": 4,
    "scoreTotal": 20,
    "subject": "二、選擇題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": "答：參見課本 ",
        "reference": "P.6-7",
        "subject": "1. 請列出三種Linux發佈的版本。",
        "type": "textarea",
      },
      {
        "answer": "答：參見課本 
( 有線網路、無線網路、行動寬頻、VPN、DSL )",
        "reference": "P.64 圖2-23",
        "subject": "2. 請列出Ubuntu中的任三種型態的網路連線設定。",
        "type": "textarea",
      },
      {
        "answer": "答：參見課本 
find: 速度慢，硬碟存取頻繁，比較耗資源
whereis 與 locate: 利用資訊庫搜尋，速度快",
        "reference": "P.139",
        "subject": "3. 在文件搜尋的指令中有 find、whereis 與 locate，請簡述在搜尋速度上有何差異?。",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 10,
    "scoreTotal": 30,
    "subject": "三、簡答題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": "答：參見課本 ",
        "reference": "P.14-15",
        "subject": "1. 安裝Ubuntu作業系統可以選擇一種稱作「Live CD」的方式，何謂「Live CD」?",
        "type": "textarea",
      },
      {
        "answer": "答：參見課本 ",
        "reference": "P.115",
        "subject": "2. 何謂命令殼層(shell)?",
        "type": "textarea",
      },
      {
        "answer": "答：參見課本 ",
        "reference": "P.128",
        "subject": "3. 請說明在終端機內指令: $ cp file1 ./subdir1/file2 ./subdir2 之意義為何 ?",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 10,
    "scoreTotal": 30,
    "subject": "四、問答題",
    "type": "section",
  },
]
`;

exports[`parse pdf string to exam data - 1042_760008_11 1`] = `
[
  {
    "subject": "國立空中大學 104 學年度下學期期中考試題",
    "type": "title",
  },
  {
    "subject": "科目：Linux 作業系統管理",
    "type": "subtitle",
  },
  {
    "children": [
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "1-1",
        "subject": "1. Linux 作業系統通常要在大型主機上使用，一般的個人電腦無法安裝使用。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "1-4-2",
        "subject": "2. 網站提供的 Ubuntu 下載檔案格式為「ISO」， ISO 格式的檔案可以直接燒錄到光碟上。",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "2-1",
        "subject": "3. Linux 作業系統並不支援圖形化的使用者介面(graphical user interface)，所以這一部分的功能跟微軟公司的視窗作業系統不同。",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "2-4-1",
        "subject": "4. IE(Internet Explorer)是 Ubuntu 系統提供的網際網路瀏覽程式。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "3-3.4",
        "subject": "5. find 指令會直接存取硬碟，比較耗費系統資源。",
        "type": "radio",
      },
    ],
    "scoreOfItem": 5,
    "scoreTotal": 25,
    "subject": "一、是非題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": "c",
        "options": [
          "(a) /dev/fd0",
          "(b) /dev/sda",
          "(c) /dev/hda",
        ],
        "reference": "1-5-2-1",
        "subject": "1. Linux 中硬碟通常對應到下列哪一種儲存裝置代號？",
        "type": "radio",
      },
      {
        "answer": "c",
        "options": [
          "(a) FAT32",
          "(b) NTFS",
          "(c) ex t4",
        ],
        "reference": "1-5-2-2",
        "subject": "2. Linux 作業系統採用過下列哪一種檔案系統格式？",
        "type": "radio",
      },
      {
        "answer": "b",
        "options": [
          "(a) 儲存裝置管理",
          "(b) 套件管理",
          "(c) 印表機的管理",
        ],
        "reference": "2-3",
        "subject": "3. Ubuntu 利用 apt-get 及 apt-cache 兩個指令的變化來進行下列哪一項操作？",
        "type": "radio",
      },
      {
        "answer": "a",
        "options": [
          "(a) 螢幕",
          "(b) 外觀",
          "(c) 主選單",
          "(d) 印表機",
        ],
        "reference": "2-1-2",
        "subject": "4. 假如要改變螢幕的解析度，在 Ubuntu 的「系統設定」中要選擇哪一項選項進行設定？",
        "type": "radio",
      },
      {
        "answer": "a",
        "options": [
          "(a) apt-get autoremove",
          "(b) apt-get remove",
          "(c) apt-get --purge remove",
        ],
        "reference": "2-3-4",
        "subject": "5. Ubuntu 中的哪一個指令可以用來移除套件，並且移除與該套件具有相依性之套件？",
        "type": "radio",
      },
      {
        "answer": "b",
        "options": [
          "(a) 本目錄",
          "(b) 家目錄",
          "(c) 主目錄",
        ],
        "reference": "3-2",
        "subject": "6. 使用者登入系統之後，內定的工作目錄稱之為？",
        "type": "radio",
      },
      {
        "answer": "c",
        "options": [
          "(a) 支援萬用字元",
          "(b) 具有設定指令別名的功能",
          "(c) 不可回溯已執行的指令",
        ],
        "reference": "3-2",
        "subject": "7. 下列有關 bash 的敘述何者錯誤？",
        "type": "radio",
      },
      {
        "answer": "c",
        "options": [
          "(a) uptime",
          "(b) vmstat",
          "(c) top",
        ],
        "reference": "3-5",
        "subject": "8. 下列哪一個指令會顯示伺服器上佔用 CPU 的任務資訊，並且每 5 秒鐘刷新一次？",
        "type": "radio",
      },
      {
        "answer": "c",
        "options": [
          "(a) pstree",
          "(b) ps",
          "(c) free",
        ],
        "reference": "3-5",
        "subject": "9. 下列哪一個指令與系統中的程序無關？",
        "type": "radio",
      },
      {
        "answer": "c",
        "options": [
          "(a) much",
          "(b) less",
          "(c) more",
        ],
        "reference": "3-3-3",
        "subject": "10. 下列哪一個指令以分頁方式顯示檔案內容，可以逐頁或逐行往後看？",
        "type": "radio",
      },
    ],
    "scoreOfItem": 5,
    "scoreTotal": 50,
    "subject": "二、選擇題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": undefined,
        "reference": "1-4-1",
        "subject": "1. 何謂「Live CD」？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "1-4-4",
        "subject": "2. Linux 作業系統與微軟 Windows 作業系統有什麼相似的地方？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "2-3-7",
        "subject": "3. Ubuntu 作業系統中使用不同的軟體來安裝軟體，會不會造成管理上的問題？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "2-4-5",
        "subject": "4. Ubuntu 作業系統中網際網路瀏覽程式需要做哪些設定才能正常使用？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "3-1-1",
        "subject": "5. 圖形介面組成要素包含哪五部分？",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 5,
    "scoreTotal": 25,
    "subject": "三、簡答題",
    "type": "section",
  },
]
`;

exports[`parse pdf string to exam data - 1122_760008_11 1`] = `
[
  {
    "subject": "國立空中大學 112 學年度下學期期中考試題",
    "type": "title",
  },
  {
    "subject": "科目：Linux 作業系統管理",
    "type": "subtitle",
  },
  {
    "children": [
      {
        "answer": "A",
        "options": [
          "(A) Ubuntu 作業系統無提供圖形化使用者介面",
          "(B) Linux 作業系統可以安裝在個人電腦或大型電腦上",
          "(C) Ubuntu 官方網站提供下載的檔案格式是「ISO」檔",
          "(D) Ubuntu Desktop 是可以安裝在個人電腦上的 Ubuntu 版本",
        ],
        "reference": "2-1",
        "subject": "1. 下列敘述何者有誤？",
        "type": "radio",
      },
      {
        "answer": "C",
        "options": [
          "(A) Firefox",
          "(B) Impress",
          "(C) 終端機",
          "(D) 設定值",
        ],
        "reference": "1-6-3",
        "subject": "2. 在 Linux 作業系統裡要以文字介面來進行簡單作業，如複製檔案或管理目錄，可以使用下列哪一個程式?",
        "type": "radio",
      },
      {
        "answer": "D",
        "options": [
          "(A) 不必再另外準備一台電腦",
          "(B) 可在一台電腦上安裝多個版本作業系統",
          "(C) 失敗多次都可以快速重來",
          "(D) 以上皆是",
        ],
        "reference": "1-4",
        "subject": "3. 下列何者是在 VirtualBox 虛擬機器軟體上安裝 Ubuntu 的好處?",
        "type": "radio",
      },
      {
        "answer": "B",
        "options": [
          "(A) 滑鼠",
          "(B) 執行中的程式與資料",
          "(C) 中央處理器",
          "(D) 檔案格式",
        ],
        "reference": "1-2-1",
        "subject": "4. 下列何者是作業系統功能裡記憶體的管理範圍?",
        "type": "radio",
      },
      {
        "answer": "C",
        "options": [
          "(A) 一般公用授權是免費的",
          "(B) 允許使用者修改和重新分發軟體的權利",
          "(C) 使用上無任何限制與規範",
          "(D) 其中一種授權狀況是不容許程式以商業化的方式使用或散佈",
        ],
        "reference": "1-1-1",
        "subject": "5. 以下關於自由軟體(Freeware)的敘述，何者錯誤?",
        "type": "radio",
      },
      {
        "answer": "C",
        "options": [
          "(A) 鍵盤",
          "(B) 電源",
          "(C) 外觀",
          "(D) 隱私",
        ],
        "reference": "2-1-3",
        "subject": "6. 要 Ubuntu 系統裡要調整桌面風格或佈景主題時，可在”設定值”裡選擇哪一個選項進行?",
        "type": "radio",
      },
      {
        "answer": "B",
        "options": [
          "(A) 根目前",
          "(B) 家目錄",
          "(C) 文件",
          "(D) 以上皆非",
        ],
        "reference": "3-2",
        "subject": "7. 下列哪一個是使用者登入 Ubuntu 系統後，內定(預設)的工作目錄？",
        "type": "radio",
      },
      {
        "answer": "D",
        "options": [
          "(A) Ubuntu 軟體中心",
          "(B) apt",
          "(C) synaptic",
          "(D) cd",
        ],
        "reference": "2-3",
        "subject": "8. 在 Ubuntu 系統裡，下列哪一個功能或指令不是用來管理或安裝套件的?",
        "type": "radio",
      },
      {
        "answer": "A",
        "options": [
          "(A) ps",
          "(B) cd",
          "(C) cp",
          "(D) ls",
        ],
        "reference": "3-3-2",
        "subject": "9. 下列何者不是與檔案與目錄管理相關的指令?",
        "type": "radio",
      },
      {
        "answer": "C",
        "options": [
          "(A) /bin",
          "(B) /root",
          "(C) /boot",
          "(D) /etc",
        ],
        "reference": "1-5",
        "subject": "10.下列哪 Ubuntu 目錄是用來存放 Linux 核心及開機程式?",
        "type": "radio",
      },
    ],
    "scoreOfItem": 5,
    "scoreTotal": 50,
    "subject": "一、選擇題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": "處理器的管理、記憶體的管理、檔案的管理、裝置的管理",
        "reference": "1-2-1",
        "subject": "1. 請寫出作業系統的四大功能。",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "1-4",
        "subject": "2. 安裝 Ubuntu 作業系統有一種方式稱作「Live CD」，請問何謂「Live CD」?",
        "type": "textarea",
      },
      {
        "answer": "穩定性、輕巧靈活、免費、多人多工的使用環境、硬體規格需求不高、多樣化圖形介面的加入",
        "reference": "1-1-4",
        "subject": "3. 請簡述 Linux 作業系統的優點與用途。",
        "type": "textarea",
      },
      {
        "answer": "桌面、視窗、選單、圖示、按鈕",
        "reference": "3-1-1",
        "subject": "4. 請列出圖形化使用者介面的五項組成要素。",
        "type": "textarea",
      },
      {
        "answer": "pwd、cd、mkdir、rmdir、ls、cp、rm、mv",
        "reference": "3-3",
        "subject": "5. 請列出至少四個與目錄或檔案相關的 Linux 指令，並簡單說明其指令功能。",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 10,
    "scoreTotal": 50,
    "subject": "二、簡答題",
    "type": "section",
  },
]
`;

exports[`parse pdf string to exam data - 1122_780022_11 1`] = `
[
  {
    "subject": "國立空中大學 112 學年度下學期期中考試題",
    "type": "title",
  },
  {
    "subject": "科目：性別、健康與多元文化",
    "type": "subtitle",
  },
  {
    "children": [
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH1 P.17",
        "subject": "1.性別運動的目的在於消除性別歧視，讓性別認同、性別特質、外表與裝扮能有多元的流動空間。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH5 P.131",
        "subject": "2.胸痛是冠狀動脈心臟病男女兩性最常見的症狀。",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH1 P.8",
        "subject": "3.女性主義理論重視性別角色與認知發展的成熟度。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": undefined,
        "subject": "4.本門課的學習目的為能了解性別、健康與多元文化的關係，學習與不同性別者的合作與接納。序",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH2 P.43",
        "subject": "5.媽祖繞境透過網路直播讓全世界認識台灣宗教文化，可視為是全球化改變了文化傳遞的方式。",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH3 P.55",
        "subject": "6.CEDAW 的全名是消除對男女一切形式歧視公約。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH4 P.87",
        "subject": "7.實施性別影響評估可以處理性別不平等的議題。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH5 P.137",
        "subject": "8.一般而言，女性骨折的風險較男性高。",
        "type": "radio",
      },
      {
        "answer": "O",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH3 P.57",
        "subject": "9.性別主流化要求公部門須將性別觀點納入政策規劃與執行及管考。",
        "type": "radio",
      },
      {
        "answer": "X",
        "options": [
          "O",
          "X",
        ],
        "reference": "CH1 P.4",
        "subject": "10.本書討論性別觀點可用性別天生一詞以蔽之。",
        "type": "radio",
      },
    ],
    "scoreOfItem": 4,
    "scoreTotal": 40,
    "subject": "一、是非題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": [
          "A",
          "C",
        ],
        "options": [
          "A 性別意識培力",
          "B 性別預算",
          "C 性別統計",
        ],
        "reference": "CH3 P.61",
        "subject": "1.2024 巴黎奧運主題之一為性別平等，也是首屆男女選手比例達 50%。請問此一進步最直接符合性別主流化哪一項工作？",
        "type": "radio",
      },
      {
        "answer": "A",
        "options": [
          "A 機會平等",
          "B 支持平等",
          "C 優惠平等",
        ],
        "reference": "CH3 P.74",
        "subject": "2.承上題 巴黎奧運也符合 CEDAW 重視的",
        "type": "radio",
      },
      {
        "answer": "B",
        "options": [
          "A 政治權",
          "B 文化權",
          "C 自治權",
        ],
        "reference": "CH2 P.30",
        "subject": "3.台灣原住民族正名運動及相關文化復興的運動，是透過何種權利的爭取而讓原住民族得到應有的尊嚴？",
        "type": "radio",
      },
      {
        "answer": "B",
        "options": [
          "A 性別統計",
          "B 性別創新",
          "C 性別意識培力",
        ],
        "reference": "CH5 P.135",
        "subject": "4.科技工程領域中性別不平等之處，進行性別分析，利用分析結果納入研發科技以利創新，藉以達到減少性別偏差，改善性別關係，促進性別平等。此稱?",
        "type": "radio",
      },
      {
        "answer": "A",
        "options": [
          "A 烹飪油煙是重要危險因子",
          "B 吸菸行為",
          "C 工廠環境暴露",
        ],
        "reference": "CH5 P.147",
        "subject": "5.亞洲非吸菸女性的肺癌環境因子暴露研究指出？",
        "type": "radio",
      },
      {
        "answer": "B",
        "options": [
          "A 譴責不平等之處",
          "B 瞭解不平等的原因",
          "C 否認性別差異",
        ],
        "reference": "CH4 P.87",
        "subject": "6.性別影響評估的目的，何者正確？",
        "type": "radio",
      },
      {
        "answer": "B",
        "options": [
          "A 重視性別多元",
          "B 反對酷兒運動",
          "C 重視論述與權力關係",
        ],
        "reference": "CH1 P.14",
        "subject": "7.下列關於第三波女性主義的敘述，何者為非？",
        "type": "radio",
      },
      {
        "answer": "C",
        "options": [
          "A 重視性別二分",
          "B 重視性別規範",
          "C 重視性別多元",
        ],
        "reference": "CH2 P.40",
        "subject": "8.從兩性關係到性別關係的敘述何者正確？",
        "type": "radio",
      },
      {
        "answer": "A",
        "options": [
          "A 全球在地化",
          "B 經濟全球化",
          "C 食物全球化",
        ],
        "reference": "CH2 P.44",
        "subject": "9.韓式炸雞近來風靡一時，韓國炸雞店也如雨後春筍般來台成立，但據報載辣度口味亦調整成台灣人可接受程度，請問此為？",
        "type": "radio",
      },
      {
        "answer": "A",
        "options": [
          "A 性別分析",
          "B 性別統計",
          "C 性別影響評估",
        ],
        "reference": "CH4P.107",
        "subject": "10.探討新生兒性別比例差異的社會文化原因，此為？",
        "type": "radio",
      },
    ],
    "scoreOfItem": 4,
    "scoreTotal": 40,
    "subject": "二、選擇題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": "答:作答者需舉例說明並呈現性別觀點對於醫療健康的重要性方可得分。",
        "reference": "CH5 P.129",
        "subject": "1.很多人認為醫療是中性的，事實上很多健康議題有其性別差異或是僅以單一性別為主的醫療研究，請舉一例說明性別觀點對於醫療健康議題的重要性。",
        "type": "textarea",
      },
      {
        "answer": "答:平等對待不同文化及族群團體成為民主國家核心的價值。多元文化的平等概念，除了消除歧視、更強調政治、社會歧視與不平等的消除，包括反對透過同化政策，進行政治社會及經濟權利的壟斷，同時給予弱勢族群較多的資源與權利保障，改善因歧視而導致的不平等。
作答者的回答只要能切合上述內容或言之成理亦可得分。",
        "reference": "CH2 P.30",
        "subject": "2.為何多元文化主義的核心內涵之一為「消除歧視與不平等對待」?請說明你的看法。",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 10,
    "scoreTotal": 20,
    "subject": "三、簡答題",
    "type": "section",
  },
]
`;

exports[`parse pdf string to exam data - 1122_780051_11 1`] = `
[
  {
    "subject": "國立空中大學 112 學年度下學期期中考試題",
    "type": "title",
  },
  {
    "subject": "科目：對聯的文學趣味",
    "type": "subtitle",
  },
  {
    "children": [
      {
        "answer": undefined,
        "reference": "(教科書第 5 頁；媒體教材 1-1-2)",
        "subject": "一、試述構成「對偶」的四項要素。",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "(教科書第 15、17 頁；媒體教材 1-2-3)",
        "subject": "二、中國在哪個時期開始在桃木板上題詞，取代了原本書寫門神名字或掛門神圖像的作法？對聯文化在哪個時代推展到民間？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "(教科書第 33、35 頁；媒體教材 1-3-5)",
        "subject": "三、南明抗清名將黃道周的諷刺聯「史筆留芳，雖未成功終可法；洪恩浩蕩，不能報國反成仇。」用了哪一個修辭手法？另外，「常將有日思無日，莫待無時思有時」又用哪個修辭手法？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "(教科書第 102-103 頁)",
        "subject": "四、清代四川總督丁寶楨十分讚許「十年宦比梅花冷，一夜春隨爆竹來」對聯，試述此聯原本旨意及其比興寓意。",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 10,
    "scoreTotal": 40,
    "subject": "壹、簡答題",
    "type": "section",
  },
  {
    "children": [
      {
        "answer": undefined,
        "reference": "(教科書第 94-96 頁；媒體教材 2-3-9)",
        "subject": "一、「異文比較」是賞析對聯的一種方式。試就「善由此地心無愧，惡過吾門膽自寒」、「善來此地心無愧，惡過我門膽自寒」說明異文含義的不同，並請賞析之。",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "(教科書第 106 頁；媒體教材 3-2-1)",
        "subject": "二、試逐項檢核並說明「大塊能相假，名山不獨藏。」是否符合基本的對聯格律？另請說明這副對聯適合張貼在什麼場所？為什麼？",
        "type": "textarea",
      },
      {
        "answer": undefined,
        "reference": "(教科書第 170-171 頁)",
        "subject": "三、試賞析江蘇無錫陸宣公祠柱聯「世上幾百年舊家，無非積德；天下第一件好事，只是讀書。」",
        "type": "textarea",
      },
    ],
    "scoreOfItem": 20,
    "scoreTotal": 60,
    "subject": "貳、申論題",
    "type": "section",
  },
]
`;
